jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_annotations.cpp

```diff
@@ -7,10 +7,61 @@
 \*******************************************************************/
 
 #include <java-testing-utils/load_java_class.h>
+#include <java-testing-utils/require_type.h>
 #include <java_bytecode/java_bytecode_convert_class.h>
 #include <java_bytecode/java_bytecode_parse_tree.h>
 #include <java_bytecode/java_types.h>
 #include <testing-utils/catch.hpp>
+#include <testing-utils/free_form_cmdline.h>
+#include <testing-utils/message.h>
+
+class test_java_bytecode_languaget : public java_bytecode_languaget
+{
+public:
+  std::vector<irep_idt> get_parsed_class_names()
+  {
+    std::vector<irep_idt> parsed_class_names;
+    for(const auto &named_class
+      : java_class_loader.get_class_with_overlays_map())
+    {
+      parsed_class_names.push_back(named_class.first);
+    }
+    return parsed_class_names;
+  }
+
+  java_class_loadert::parse_tree_with_overlayst &get_parse_trees_for_class(
+    const irep_idt &class_name)
+  {
+    return java_class_loader.get_class_with_overlays_map().at(class_name);
+  }
+};
+
+static irep_idt get_base_name(const typet &type)
+{
+  return type.get(ID_C_base_name);
+}
+
+static void require_matching_annotations(
+  const java_bytecode_parse_treet::annotationst &annotations,
+  const std::set<irep_idt> &expectedAnnotations)
+{
+  REQUIRE(annotations.size() == expectedAnnotations.size());
+  std::set<irep_idt> annotation_names;
+  std::transform(
+    annotations.begin(),
+    annotations.end(),
+    std::inserter(annotation_names, annotation_names.end()),
+    [&annotations](const java_bytecode_parse_treet::annotationt &annotation)
+    {
+      return get_base_name(
+        require_type::require_pointer(annotations[0].type, {}).subtype());
+    });
+  auto result = std::mismatch(
```

Comment:
:pick: Now you've done the above transform, I think you can use:
```C++
REQUIRE_THAT(
      annotation_names, 
      Catch::Matchers::Vector::EqualsMatcher<irep_idt>{expected_annotations});
```
and the output from the unit test will even tell you which ones don't match (you can also ditch the size check at the top unless it is needed for the transform to work?)